exercises/robot-simulator/src/robot_simulator.h

```diff
@@ -0,0 +1,40 @@
+#ifndef _ROBOT_SIMULATOR_H
+#define _ROBOT_SIMULATOR_H
+
+typedef enum {
+   HeadingNorth = 0,
+   HeadingEast,
+   HeadingSouth,
+   HeadingWest,
+   HeadingMax
+} Bearing_t;
+
+enum {
+   Default_Bearing = HeadingNorth,
+   Default_X_Coordinate = 0,
+   Default_Y_Coordinate = 0,
+};
+
+enum {
+   Command_Left = 'L',
+   Command_Right = 'R',
+   Command_Advance = 'A'
+};
+
+typedef struct RobotCoordinates {
+   int x_position;
+   int y_position;
+} Robott_Coordinates_t;
+
+typedef struct RobotGridStatus {
+   Bearing_t bearing;
+   Robott_Coordinates_t grid;
+} RobotGridStatus_t;
+
+RobotGridStatus_t robot_init(int args, ...);
```

Comment:
I agree, this approach seems more complex than necessary. +1 for two `init` functions

That will help make it clear what the interface is supposed to be. The tests mostly explain it, but it could be better. For example, since `Robot_Coordinates_t` is defined I would have expected this function definition to be something like:

```
RobotGridStatus_t robot_init(Bearing_t bearing, Robot_Coordinates_t grid);
```